nemo/collections/asr/models/rnnt_models.py

```diff
@@ -699,6 +698,23 @@ def training_step(self, batch, batch_nb):
 
         return {'loss': loss_value}
 
+    def predict_step(self, batch, batch_idx, dataloader_idx=0):
+        signal, signal_len, transcript, transcript_len, sample_id = batch
+
+        # forward() only performs encoder forward
+        if isinstance(batch, DALIOutputs) and batch.has_processed_signal:
+            encoded, encoded_len = self.forward(processed_signal=signal, processed_signal_length=signal_len)
+        else:
+            encoded, encoded_len = self.forward(input_signal=signal, input_signal_length=signal_len)
+        del signal
+
+        best_hyp_text, all_hyp_text = self.decoding.rnnt_decoder_predictions_tensor(
```

Comment:
For RNNTs, you can have beam decoding (i think its too slow but still an option). So better to return hypotheses = True and return the hypotheses, all_hypotheses as output.

Reply:
Added error message in the dali layers that they do not support return_sample_id. Here is a general predict which works with or without return_sample_id and may be used in other cases without returning the index.

Reply:
Cool thanks.